### PR TITLE
Add local database command with event fetching functionality

### PR DESCRIPTION
**Pull request by findolor**
> [!CAUTION]
> **Chained PR** - Do not merge before https://github.com/rainlanguage/rain.orderbook/pull/2101

## Motivation

This change introduces local database operations to the CLI, specifically focusing on
  blockchain event fetching capabilities.

## Solution

  Added a new local-db command module with the following components:

  - New CLI command structure: Added local-db fetch-events subcommand that allows users
  to fetch events from blockchain networks and save them to JSON files
  - Dependency injection pattern: Implemented an EventClient trait to enable testable
  code with mock clients for unit testing
  - Hyper RPC client enhancements: Extended the HyperRpcClient to support Arbitrum (chain
   ID 42161) in addition to Base, and exposed the client through SqliteWeb for dependency
   injection
  - Comprehensive testing: Added extensive unit tests covering success cases, error
  handling, and edge cases using mock implementations
  - File management: Added appropriate .gitignore entries to exclude generated event
  output files from version control

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

* **New Features**
  * Added a local-db CLI group with fetch-events and decode-events subcommands.
  * fetch-events: fetch blockchain events for an address over a block range (end defaults to latest) and save pretty JSON (custom path or via RAIN_ORDERBOOK_EVENTS_DIR).
  * decode-events: decode events from a JSON file and write decoded output (defaults when not provided).
  * RPC connections now support Arbitrum (chain ID 42161).

* **Chores**
  * Ignore generated event JSON files.
  * Adjusted CLI dependency declarations for builds.

### CODE REVIEW

**Review by hardyjosh — APPROVED** [review]

File: crates/common/src/hyper_rpc.rs [hunk 1]
```diff
@@ -78,11 +78,14 @@ impl std::fmt::Debug for HyperRpcClient {
...
 impl HyperRpcClient {
     pub fn new(chain_id: u32, api_token: String) -> Result<Self, HyperRpcError> {
-        let rpc_url = match chain_id {
-            8453 => format!("https://base.rpc.hypersync.xyz/{}", api_token),
+        let base_url = match chain_id {
+            8453 => "https://base.rpc.hypersync.xyz".to_string(),
+            42161 => "https://arbitrum.rpc.hypersync.xyz".to_string(),
             _ => return Err(HyperRpcError::UnsupportedChainId { chain_id }),
         };
+        let rpc_url = format!("{}/{}", base_url, api_token);
```
Comment (0xgleb) [inline crates/common/src/hyper_rpc.rs:86]:
```suggestion
        let rpc_url = format!("{base_url}/{api_token}");
```
